pulp_rpm/tests/functional/api/test_download_policies.py

```diff
@@ -69,51 +59,39 @@ def do_sync(self, download_policy):
         4. Assert that repository version is not None.
         5. Assert that the correct number of possible units to be downloaded
            were shown.
-        6. Sync the remote one more time.
-        7. Assert that repository version is the same as the previous latest version.
+        6. Sync the remote one more time in order to create another repository
+           version.
+        7. Assert that repository version is different from the previous one.
+        8. Assert that the same number of units are shown, and after the
+           second sync no extra units should be shown, since the same remote
+           was synced again.
         """
-        repo = self.client.post(RPM_REPO_PATH, gen_repo())
-        self.addCleanup(self.client.delete, repo['pulp_href'])
+        repo_api = RepositoriesRpmApi(self.client)
+        remote_api = RemotesRpmApi(self.client)
+
+        repo = repo_api.create(gen_repo())
+        self.addCleanup(repo_api.delete, repo.pulp_href)
 
-        remote = self.client.post(
-            RPM_REMOTE_PATH,
-            gen_rpm_remote(policy=download_policy)
-        )
-        self.addCleanup(self.client.delete, remote['pulp_href'])
+        body = gen_rpm_remote(**{"policy": download_policy})
+        remote = remote_api.create(body)
+        self.addCleanup(remote_api.delete, remote.pulp_href)
 
         # Sync the repository.
-        self.assertEqual(repo["latest_version_href"], f"{repo['pulp_href']}versions/0/")
-        sync(self.cfg, remote, repo)
-        repo = self.client.get(repo['pulp_href'])
+        self.assertEqual(repo.latest_version_href, f"{repo.pulp_href}versions/0/")
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+        repo = repo_api.read(repo.pulp_href)
 
-        self.assertIsNotNone(repo['latest_version_href'])
-        self.assertDictEqual(get_content_summary(repo), RPM_FIXTURE_SUMMARY)
-        self.assertDictEqual(
-            get_added_content_summary(repo),
-            RPM_FIXTURE_SUMMARY
-        )
+        self.assertIsNotNone(repo.latest_version_href)
+        self.assertDictEqual(get_content_summary(repo.to_dict()), RPM_FIXTURE_SUMMARY)
+        self.assertDictEqual(get_added_content_summary(repo.to_dict()), RPM_FIXTURE_SUMMARY)
 
         # Sync the repository again.
-        latest_version_href = repo['latest_version_href']
-        sync(self.cfg, remote, repo)
-        repo = self.client.get(repo['pulp_href'])
-
-        self.assertEqual(latest_version_href, repo['latest_version_href'])
-        self.assertDictEqual(get_content_summary(repo), RPM_FIXTURE_SUMMARY)
-
-    def do_publish(self, download_policy):
```

Comment:
the original test case also tests publish, so it needs to be converted also

Reply:
publish is joined sync in one 'do_test' method